packages/react/src/metadata.tsx

```diff
@@ -345,7 +345,7 @@ export const useActionMetadata = (actionFunction: ActionFunction<any, any, any,
  */
 export const filterAutoFormFieldList = (
   fields: FieldMetadata[] | undefined,
-  options?: { include?: string[]; exclude?: string[] }
+  options?: { include?: string[]; exclude?: string[]; allowIdFieldType?: boolean }
```

Comment:
I think I'd prefer this option to refer to the fact that it's related to upserting -- we've fallen into a bad happen of passing these behavior control flags around which can be difficult to understand the implications of changing down the road. 

Maybe this would be `options?: { include?: string[]; exclude?: string[]; upsertAction?: boolean }`

Reply:
Updated such that the option name refers to the upsert operation